src/gui/widgets/FloatModelEditorBase.cpp

```diff
@@ -278,8 +282,79 @@ void FloatModelEditorBase::paintEvent(QPaintEvent *)
 	p.setPen(foreground);
 	p.setBrush(foreground);
 	p.drawRect(QRect(r.topLeft(), QPoint(r.width() * percentage, r.height())));
+	drawAutoHighlight(&p);
 }
 
+std::vector<InteractiveModelView::ModelShortcut> FloatModelEditorBase::getShortcuts()
+{
+	std::vector<InteractiveModelView::ModelShortcut> shortcuts = {
+		InteractiveModelView::ModelShortcut(Qt::Key_C, Qt::ControlModifier, 0, QString(tr("Copy value")), false),
+		InteractiveModelView::ModelShortcut(Qt::Key_C, Qt::ControlModifier, 1, QString(tr("Link widget")), false),
+		InteractiveModelView::ModelShortcut(Qt::Key_V, Qt::ControlModifier, 0, QString(tr("Paste value")), false)
+	};
+	return shortcuts;
+}
+
+void FloatModelEditorBase::processShortcutPressed(size_t shortcutLocation, QKeyEvent* event)
+{
+	switch (shortcutLocation)
+	{
+		case 0:
+			Clipboard::copyStringPair(Clipboard::StringPairDataType::FloatValue, Clipboard::clipboardEncodeFloatValue(model()->value() * getConversionFactor()));
+			InteractiveModelView::startHighlighting(Clipboard::StringPairDataType::FloatValue);
+			break;
+		case 1:
+			Clipboard::copyStringPair(Clipboard::StringPairDataType::AutomatableModelLink, Clipboard::clipboardEncodeAutomatableModelLink(model()->id()));
+			InteractiveModelView::startHighlighting(Clipboard::StringPairDataType::AutomatableModelLink);
+			break;
+		case 2:
+			processPaste(Clipboard::getMimeData());
+			break;
+		default:
+			break;
+	}
+}
+
+QString FloatModelEditorBase::getShortcutMessage()
+{
+	return m_shortcutMessage;
+}
+
+bool FloatModelEditorBase::canAcceptClipboardData(Clipboard::StringPairDataType dataType)
+{
+	return dataType == Clipboard::StringPairDataType::FloatValue
+		|| dataType == Clipboard::StringPairDataType::AutomatableModelLink;
+}
+
+bool FloatModelEditorBase::processPaste(const QMimeData* mimeData)
+{
+	if (Clipboard::hasFormat(Clipboard::MimeType::StringPair) == false) { return false; }
+	bool shouldAccept = false;
+
+	Clipboard::StringPairDataType type = Clipboard::decodeKey(mimeData);
+	QString value = Clipboard::decodeValue(mimeData);
```

Comment:
To avoid some of this boilerplate in classes that implement `InteractiveModelView`, maybe this method could have a signature like this:
```c++
bool processPasteImpl(Clipboard::DataType type, const QString& value)
```

Or whatever you want to call it.
The `InteractiveModelView` class would implement the mime data decoding in a non-virtual `processPaste` method which then calls the pure virtual `processPasteImpl` which is implemented in derived classes. `processPaste` could also handle the potential  `InteractiveModelView::stopHighlighting();` call after `processPasteImpl` returns.

The idea is to separate the concerns, reduce duplicate code, and make it really easy for a derived class to implement `InteractiveModelView`.